src/components/toast-notification-manager/daikin-toast-notification-manager.ts

```diff
@@ -0,0 +1,233 @@
+
+  @queryAssignedElements({ selector: "daikin-toast-notification" })
+  private readonly _items!: readonly DaikinToastNotification[];
+
+  @state()
+  private _beforeItems: readonly DaikinToastNotification[] = [];
+
+  @state()
+  _newItem: DaikinToastNotification | null = null;
+
+  private _containerRef = createRef<HTMLElement>();
+
+  get _positionY(): "top" | "bottom" {
+    return this.position.startsWith("top") ? "top" : "bottom";
+  }
+
+  private _close(target: DaikinToastNotification) {
+    const items = this._items;
+    const targetIndex = items.findIndex((item) => item === target);
+    const afterItems = items.filter((_, index) =>
+      this._positionY === "top" ? index > targetIndex : index < targetIndex
+    );
+
+    for (const item of items) {
+      item.style.removeProperty(`--transition-duration`);
+    }
+
+    target.style.setProperty(`--move-offset-x`, TOAST_MOVE_OFFSET_X);
+    target.style.setProperty(`--opacity`, "0");
+    target.style.setProperty(`--pointer-events`, "none");
+
+    for (const item of afterItems) {
+      const height = item.clientHeight;
+      item.style.setProperty(
+        `--move-offset-y`,
+        `calc(0.5rem + ${
+          height *
+          (this._positionY === "top"
+            ? TOAST_MOVE_OFFSET_Y_TOP_SIGN
+            : TOAST_MOVE_OFFSET_Y_BOTTOM_SIGN)
+        }px)`
+      );
+    }
+
+    setTimeout(() => {
+      for (const item of this._items) {
+        item.style.setProperty(`--transition-duration`, "0");
+      }
+
+      this.dispatchEvent(
+        new CustomEvent("close", { detail: { name: target.name } })
+      );
+      for (const item of this._items) {
+        item.style.removeProperty(`--move-offset-y`);
+      }
+    }, TOAST_DURATION);
+  }
+
+  private _handleClose(event: Event) {
+    event.stopPropagation();
+
+    this._close(event.target as DaikinToastNotification);
+  }
+
+  private _handleSlotchange() {
+    const items = this._items;
+
+    this._newItem =
+      items.find(
+        (item) => !this._beforeItems.find((beforeItem) => item === beforeItem)
+      ) ?? null;
+    this._beforeItems = items;
+  }
+
+  private _removeToast(target: DaikinToastNotification) {
+    if (!this.duration) {
+      return;
+    }
+
+    setTimeout(() => {
+      if (!this._items.find((item) => item.name === target.name)) {
+        return;
+      }
+
+      this._close(target);
+    }, this.duration);
+  }
+
+  override render() {
+    return html`<div
+      ${ref(this._containerRef)}
+      class=${cvaContainer({ position: this.position })}
+      style=${`transform:translateY(var(${CONTAINER_MOVE_OFFSET_Y},0));transition-duration:var(${CONTAINER_TRANSITION_DURATION},${TOAST_DURATION}ms);`}
+    >
+      <slot
+        @close=${this._handleClose}
+        @slotchange=${this._handleSlotchange}
+      ></slot>
+    </div>`;
+  }
+
+  protected override updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has("_newItem") && !!this._newItem) {
```

Comment:
Do 

```suggestion
  private _handleNewItem(item: DaikinToastNotification): void {
```

instead of `updated` lifecycle, as we know when we have to handle new items.